### PR TITLE
#50 - All Transforms

### PR DESCRIPTION
**Pull request by kalebphipps**
# Description
This PR includes the additional transformations required for zero-shot testing etc. This resolves Issue #50.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by oskar-taubert — COMMENTED** [review]
I'll pass this on to @JuanPedroGHM

File: VOLTA/data/transforms/__init__.py [hunk 1]
```diff
@@ -1,16 +1,66 @@
...
     RandomEdgeMasking,
     RandomNodeMasking,
+    TimeFeatureMaskingForEdges,
+    TimeFeatureMaskingForNodes,
+    TimeMaskingForEdges,
+    TimeMaskingForNodes,
 )
 
 __all__ = [
     "VoltaBaseTransform",
```
Comment (JuanPedroGHM) [inline VOLTA/data/transforms/__init__.py:36]: Maybe rename this to "BaseTransform" or just "Transform". To keep it consistent with the rest of the code base.

File: VOLTA/data/transforms/base_transform.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def forward(self, data: Any, mask: Any) -> Tuple[Any, Any]:
         data : Any
             The transformed data.
         mask : Any
-            The accumulated mask of the transformations.
+            The accumulated mask of all the transformations.
```
Comment (JuanPedroGHM) [inline VOLTA/data/transforms/base_transform.py:47]: Is there a reason why BaseTransform shouldn't also take torch_geometric.data.Data objects as input and output?

File: VOLTA/data/transforms/distorting.py [hunk 3]
```diff
@@ -137,3 +137,807 @@ def forward(
         mask.edge_attr *= distortion_mask
 
         return data, mask
+
+
+@functional_transform("time_distortion_for_node")
+class TimeDistortionForNodes(VoltaBaseTransform):
+    """
+    Distort time with given node and feature.
```
[the remark this answers is not in the document]
Comment (JuanPedroGHM) [reply VOLTA/data/transforms/distorting.py:148]: I agree with this. Examples on the docstrings could help illustrate what is going on with each version.

File: VOLTA/data/transforms/masking.py [hunk 4]
```diff
@@ -127,3 +127,807 @@ def forward(self, data: Data, mask: Data) -> Tuple[Data, Data]:
         mask.edge_attr *= masking_mask
 
         return data, mask
+
+
+@functional_transform("time_masking_for_nodes")
+class TimeMaskingForNodes(VoltaBaseTransform):
```
Comment (JuanPedroGHM) [inline VOLTA/data/transforms/masking.py:133]: The same applies here. Docstrings are too similar. The class names could also use some reworking. The difference between "NodeFeatureMasking" and "FeatureMaskingForNodes" is not clear.

**Review by JuanPedroGHM — CHANGES_REQUESTED** [review]
Some suggestions:
 - Transformations should be more easily distinguishable (trough name and docstrings).
 - The `forward` method signature should be consistent across all the subclasses. 
 - Missing tests for edge cases, and different types of data, and different Transformation arguments. (Will make an issue for it, no need to change it here, as this is a repository problem in general)
